SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -1,130 +1,433 @@
+                measures_package = new[]
+                {
+                    "A", "A2", "B", "Q", "Q1", "W1", "W2", "D", "C", "F", "G", "I", "T", "L2", "N", "Y", "O", "O3", "X",
+                    "U"
+                }
             };
 
-            // Glazing is single or both
-            if (userData.GlazingType is GlazingType.Both or GlazingType.SingleGlazed)
+            return request;
+        }
+
+        private static int GetConvertedPropertyType(PropertyType? propertyType)
+        {
+            //ApartmentFlatOrMaisonette is assumed to be a Flat
+            if (propertyType != null) return (int) (PropertyTypeEnum) propertyType;
+            throw new ArgumentNullException();
+        }
+
+        private static (int builtForm, int? flatLevel) GetConvertedBuiltFormAndFlatLevel(PropertyType? propertyType,
+            HouseType? houseType, BungalowType? bungalowType, FlatType? flatType)
+        {
+            int builtForm;
+            int? flatLevel = null;
+            switch (propertyType)
             {
-                userRecommendationKeys.Add(RecommendationKey.FitNewWindows);
+                case PropertyType.House:
+                    builtForm = houseType switch
+                    {
+                        HouseType.Detached => (int) BuiltFormEnum.Detached,
+                        HouseType.SemiDetached => (int) BuiltFormEnum.SemiDetached,
+                        //assumption:
+                        HouseType.EndTerrace => (int) BuiltFormEnum.EndTerrace,
+                        //assumption:
+                        HouseType.Terraced => (int) BuiltFormEnum.MidTerrace,
+                        _ => throw new ArgumentOutOfRangeException()
+                    };
+                    break;
+                case PropertyType.Bungalow:
+                    builtForm = bungalowType switch
+                    {
+                        BungalowType.Detached => (int) BuiltFormEnum.Detached,
+                        BungalowType.SemiDetached => (int) BuiltFormEnum.SemiDetached,
+                        //assumption:
+                        BungalowType.EndTerrace => (int) BuiltFormEnum.EndTerrace,
+                        //assumption:
+                        BungalowType.Terraced => (int) BuiltFormEnum.MidTerrace,
+                        _ => throw new ArgumentOutOfRangeException()
+                    };
+                    break;
+                case PropertyType.ApartmentFlatOrMaisonette:
+                    flatLevel = flatType switch
+                    {
+                        FlatType.TopFloor => (int) FlatLevelEnum.TopFloor,
+                        FlatType.MiddleFloor => (int) FlatLevelEnum.MidFloor,
+                        FlatType.GroundFloor => (int) FlatLevelEnum.GroundFloor,
+                        _ => throw new ArgumentOutOfRangeException()
+                    };
+                    //assumption:
```

Comment:
I think this needs a better comment e.g. `builtForm doesn't make sense for a flat, but is currently a mandatory field on the BRE API so we set MidTerrace as a default`